src/ralph/operations/models.py

```diff
@@ -0,0 +1,145 @@
+        db_index=True
+    )
+
+    class choices(Choices):
+        _ = Choices.Choice
+
+        CHANGES = Choices.Group(0)
+        change = _('change')
+
+        INCIDENTS = Choices.Group(100)
+        incident = _('incident')
+
+        PROBLEMS = Choices.Group(200)
+        problem = _('problem')
+
+        FAILURES = Choices.Group(300)
+        failure = _('failure')
+        hardware_failure = _('hardware failure')
+
+
+class Operation(AdminAbsoluteUrlMixin, TaggableMixin, models.Model):
+    type = TreeForeignKey(OperationType, verbose_name=_('type'))
+    title = models.CharField(
+        max_length=350, null=False, blank=False, verbose_name=_('title'),
+    )
+    description = models.TextField(
+        verbose_name=_('description'), null=True, blank=True,
+    )
+    status = models.PositiveIntegerField(
+        verbose_name=_('status'), choices=OperationStatus(),
+    )
+    asignee = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        related_name='operations',
+        verbose_name=_('asignee'),
+        null=True,
+        blank=True,
+    )
+    ticket_id = NullableCharField(
+        verbose_name=_('ticket id'),
+        help_text=_('External system ticket identifier'),
+        null=True, blank=True,
+        max_length=20,
+    )
+    created_date = models.DateTimeField(
+        null=True, blank=True, verbose_name=_('created date'),
+    )
+    update_date = models.DateTimeField(
+        null=True, blank=True, verbose_name=_('updated date'),
+    )
+    resolved_date = models.DateTimeField(
+        null=True, blank=True, verbose_name=_('resolved date'),
+    )
+    base_objects = models.ManyToManyField(
+        BaseObject, related_name='operations', verbose_name=_('objects'),
+        blank=True,
+    )
+
+    def __str__(self):
+        return self.title
+
+
+class OperationDescendantManager(models.Manager):
+    """
+    Narrow Operations only to particular type.
+    """
+    def __init__(self, descendants_of, *args, **kwargs):
+        self._descendants_of = descendants_of
+        super().__init__(*args, **kwargs)
+
+    def get_queryset(self, *args, **kwargs):
+        queryset = super().get_queryset(*args, **kwargs)
+        root = OperationType.objects.get(pk=self._descendants_of)
+        return queryset.filter(type__in=root.get_descendants(include_self=True))
+
+
+class Change(Operation):
+    objects = OperationDescendantManager(OperationType.choices.change)
+
+    class Meta:
+        proxy = True
+
+
+class Incident(Operation):
+    objects = OperationDescendantManager(OperationType.choices.incident)
+
+    class Meta:
+        proxy = True
+
+
+class Problem(Operation):
+    objects = OperationDescendantManager(OperationType.choices.problem)
+
+    class Meta:
+        proxy = True
+
+
+class Failure(Operation):
+    objects = OperationDescendantManager(OperationType.choices.failure)
+
+    class Meta:
+        proxy = True
+
+
+@receiver(post_migrate)
+def rebuild_handler(sender, **kwargs):
```

Comment:
in migration 0002 we set `{'lft': 0, 'rght': 0, 'level': 0, 'tree_id': 0}`, so after this migration (and possibly every other migration in the future which will add more types) we have to rebuild mptt tree.